pagopa/io-app
# chore: [IOPID-243] Add local and remote FF for fast and native login

## Short description
This PR adds the management of local and remote FF for fast and native login.

The selectors used for FF use a `PatternString` decoding that fails for non standard `SemVer` and for all `SemVer` of all `0`s (`0`, `0.0`, `0.0.0`, `0.0.0.0`).

## List of changes proposed in this pull request
- [.env.local](https://github.com/pagopa/io-app/pull/4651/files#diff-74dd9d08974aed4457940214a64b43ce175d062d74e72309b003f02c24f98b30): Adds `enabled` local flags.
- [.env.production](https://github.com/pagopa/io-app/pull/4651/files#diff-48f73ea2653e45c41f31afc2d42dfde14d8caf26d8dd24d3d2a81a52604f4cb6): Adds `disabled` local flags.
- [package.json](https://github.com/pagopa/io-app/pull/4651/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519): Bumps metadata tag from `1.0.24` to `1.0.25`.
- [ts/config.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-0f5e4629af7b77f0b21af3a54fe1d4d21314eb44c8a54193316d9d80ba2654bf): Adds local flags.
- [ts/features/fastLogin/store/selectors/index.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-77b2bc00cdd44900b86f533de9771b38b7d79e0e7dbfedf5b3119ca11b1d6eba): Adds fast login FF selector.
- [ts/features/fastLogin/store/selectors/__tests__/fastLogin.test.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-4184bb1097ce53f8a3e5f22be9786a1303502e948d1c3016c17bb6cc7ef184f6): Adds fast login FF selector tests.
- [ts/features/nativeLogin/store/selectors/index.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-5da0f6684e8f402fce1941a4ed4bde7c83d04e2c7769fdda96f9d61e8adfcaf6): Adds native login FF selector.
- [ts/features/nativeLogin/store/selectors/__tests__/nativeLogin.test.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-b7193eff79bd4f250120cd35c1623de04b1d1212d4a1b1aa0ca449f9c9b9823a): Adds native login FF selector tests.
- [ts/store/reducers/backendStatus.ts](https://github.com/pagopa/io-app/pull/4651/files#diff-164bd382961cc4552dea2afabb89bc07bf7c2b20e3ab109f344be2474fa40b1a): Export an utility function to be used with those FF with `min_app_version` property..

## How to test
See that all checks are passed. Then run the app against [this dev server PR](https://github.com/pagopa/io-dev-api-server/pull/258) and try different values of [`min_app_version`](https://github.com/pagopa/io-dev-api-server/blob/482d4da8e19d49a0a0caaebe9942e4b53ca6f1d9/src/payloads/backend.ts#L94). Check that the corresponding selectors work accordingly.
